src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/ResourceFormStore.js

```diff
@@ -87,15 +87,139 @@ export default class ResourceFormStore extends AbstractFormStore implements Form
         });
     };
 
-    @action handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
+    handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
         this.validator = jsonSchema ? ajv.compile(jsonSchema) : undefined;
         this.pathsByTag = {};
 
-        this.schema = schema;
-        this.addMissingSchemaProperties();
-        this.setSchemaLoading(false);
+        return this.loadAndMergeRemoteData(this.schema, schema).then(action(() => {
+            this.schema = schema;
+            this.addMissingSchemaProperties();
+            this.validate();
+            this.setSchemaLoading(false);
+        }));
     };
 
+    loadAndMergeRemoteData = (localSchema: Schema, remoteSchema: Schema) => {
+        // load data only after initial schema was set to prevent duplicate requests during initialization
+        if (localSchema) {
+            return this.resourceStore.requestRemoteData({template: this.type}).then((data: Object) => {
+                const result = this.mergeData(localSchema, remoteSchema, this.data, data);
+                this.setMultiple(result);
+            });
+        }
+        return Promise.resolve();
+    };
+
+    mergeData(
```

Comment:
think i would define this method outside of the class like we did in the `AbstractFormStore`: https://github.com/sulu/sulu/blob/5e4f11ac8d160c4a1242b8454265925c72a9cf30/src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/AbstractFormStore.js#L10-L25